resolwe_bio/tools/expressionmerge_archive.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+"""Merge columns of multiple experiments by gene id."""
+
+import argparse
+import csv
+import os
+import utils
+
+import pandas as pd
+import numpy as np
+
+parser = argparse.ArgumentParser(description="Merge columns of multiple experiments by gene id.")
+
+parser.add_argument('-f', '--input_file', required=True, help="Txt file with paths to expression files.")
+
+args = parser.parse_args()
+
+
+def merge_expressions(input_file):
+    """Merge expressioms."""
+    df = pd.DataFrame(np.nan, index=[], columns=[])
+    headers = []
+
+    with open(input_file, 'r') as tfile:
+        for line in tfile:
+            line = line.rstrip()
+            header = os.path.basename(line).rstrip()
+            if header.endswith('.tab.gz'):
+                header = header[:-7]
+            headers.append(header)
+
+            with utils.gzopen(line) as csvfile:
+                reader = pd.read_csv(csvfile, index_col='Gene', delimiter='\t', dtype=str)
+                df = pd.concat([df, reader], axis=1)
+    df.columns = headers
+    return df
+
+
+def write_output(input_file, df):
+    """Compose a name and write output file."""
+    with open(input_file, 'r') as tfile:
+        # get Species and Build from input file and use it for path of output file
+        first_line = tfile.readline().strip()
```

Comment:
Rewrite this like so::

    first_line = next(tfile).strip()
    # Line is a path of the following form: 
    # species/build/sample_slug/filetype/filename
    species, build, _, _, _ = os.path.normpath(first_line).split(os.path.sep)